third_party/terraform/resources/resource_bigquery_table.go.erb

```diff
@@ -838,6 +907,28 @@ func expandTimePartitioning(configured interface{}) *bigquery.TimePartitioning {
 	return tp
 }
 
+<% unless version == 'ga' -%>
+func expandRangePartitioning(configured interface{}) (*bigquery.RangePartitioning, error) {
```

Comment:
we had a couple of type changes so it looks like this expander doesn't work - wrote up an example function for what it should be afterwards

```
func expandRangePartitioning(configured interface{}) (*bigquery.RangePartitioning, error) {
	if configured == nil {
		return nil, nil
	}

	ls := configured.([]interface{})
	if len(ls) == 0 || ls[0] == nil {
		return nil, errors.New("Error casting range partitioning interface to expected structure")
	}

	original := ls[0].(map[string]interface{})
	transformed := &bigquery.RangePartitioning{
		Field: original["field"].(string),
	}

	if v, ok := original["range"]; ok && v != nil {
		rangeLs := v.([]interface{})
		if len(rangeLs) != 1 || rangeLs[0] == nil {
			return nil, errors.New("non-empty range must be given for range partitioning")
		}

		originalRange := rangeLs[0].(map[string]interface{})
		transformed.Range = &bigquery.RangePartitioningRange{
			Start:    int64(originalRange["start"].(int)),
			End:      int64(originalRange["end"].(int)),
			Interval: int64(originalRange["interval"].(int)),
		}
	}
	return transformed, nil
}
```